WCF/Shared/Implementation/WcfOperationContext.cs

```diff
@@ -115,7 +129,7 @@ private static IOperationContext GetContext()
             {
                 if ( owner != null )
                 {
-                    context = new WcfOperationContext(owner);
+                    context = new WcfOperationContext(owner, HttpContext.Current);
```

Comment:
That could work, though to be more exact, we'd only care about the `HttpContext` if we're running under IIS *and* the service is called over HTTP (though not sure if that check would be more expensive in the end).